geopyspark/geotrellis/__init__.py

```diff
@@ -539,6 +539,48 @@ def __new__(cls, num_partitions=None, bits=8):
         return super(cls, SpatialPartitionStrategy).__new__(cls, num_partitions, bits)
 
 
+class SpaceTimePartitionStrategy(namedtuple("SpaceTimePartitionStrategy", "time_unit num_partitions bits time_resolution")):
+    """Represents a partitioning strategy for a layer that uses GeoPySpark's ``SpaceTimePartitioner``
+    with a set number of partitions, units of time, and temporal resolution.
+
+    This partitioner will try and group ``Tile``\s together that are spatially and temproally near each other
+    in the same partition. In order to do this, each ``Tile`` has their ``Key Index`` calculated
+    using the space filling curve index, ``Z-Curve``.
+
+    Note:
+        This partitiong strategy will only work on ``SPACETIME`` layers, and will fail if given a ``SPATIAL``
+        one. For ``SPATIAL`` layers, please see :class:`~geopyspark.geotrellis.SpatialPartitionStrategy`.
+
+    Args:
+        time_unit (str or :class:`~geopyspark.geotrellis.constants.TimeUnit`): Which time
+            unit should be used when saving spatial-temporal data. This controls the resolution of
+            each index. Meaning, what time intervals are used to seperate each record.
+        num_partitions (int, optional): The number of partitions that should be used during
+            partitioning. Default is, ``None``. If ``None`` the resulting layer will have
+            a ``HashPartitioner`` with the number of partitions being either the same
+            as the input layer's, or a number computed by the method.
+        bits (int, optional): How many bits the resulting ``Key Index`` of a ``Tile`` should
+            be shifted to the right. Default is, ``8``.
+        time_resolution (str or int, optional): Determines how data for each ``time_unit`` should be
+            grouped together. By default, no grouping will occur.
+
+            As an example, having a ``time_unit`` of ``WEEKS`` and a ``time_resolution`` of 5 will
+            cause the data to be grouped and stored together in units of 5 weeks. If however
+            ``time_resolution`` is not specified, then the data will be grouped and stored in units
+            of single weeks.
+
+            This value can either be an ``int`` or a string representation of an ``int``.
+
+    Returns:
+        :class:`~geopyspark.geotrellis.SpaceTimePartitionStrategy`
+    """
+
+    __slots__ = []
+
+    def __new__(cls, time_unit, num_partitions=None, bits=8, time_resolution=None):
```

Comment:
One thing that is worth noting here is that the order of the parameters for `SpaceTimePartitionStrategy` are in a slightly different order than that of `SpatialPartitionStrategy`. This is because `time_unit` is the only required parameter whereas all of the others are optional, and it makes more sense to me to have the required parameter be first.